05_cleaning_data_04_reshapingdata.Rmd

```diff
@@ -0,0 +1,222 @@
+
+```r
+## to rename the column names that gather provides,
+## change key and value to what you want those column names to be
+gathered <- gather(airquality, key="variable", value="value")
+
+## take a look at first few rows of long data
+head(gathered)
+```
+
+
+![gather column names changed](https://docs.google.com/presentation/d/14msuN3MbQE6BSIaNu2ipv1-5ypgvWlxsGwn3jmpFyAI/export/png?id=14msuN3MbQE6BSIaNu2ipv1-5ypgvWlxsGwn3jmpFyAI&pageid=g38bb68a532_0_213)
+
+However, you're likely not interested in your day and month variable being separated out into their own variables within the `key` column. In fact, knowing the day and month associated with a particular data point helps identify that particular data point. To account for this, you can exclude `day` and `month` from the variables being included in the `key` column by specifying all the variables that you *do* want included in the `key` column.  Here, that means specifying `ozone`, `solar.r`, `wind`, and `temp`. This will keep `day` and `month` in their own columns, allowing each row to be identified by the specific day and month being discussed.
+
+```r
+## in gather(), after key and value, you can specify which variables
+## you want included in the long format
+## it will leave the other variables as is
+gathered <- gather(airquality, key="variable", value="value", ozone, solar.r, wind, temp)
+
+## take a look at first few rows of long data
+head(gathered)
+```
+
+
+![gather specifying which variables to include in long format](https://docs.google.com/presentation/d/14msuN3MbQE6BSIaNu2ipv1-5ypgvWlxsGwn3jmpFyAI/export/png?id=14msuN3MbQE6BSIaNu2ipv1-5ypgvWlxsGwn3jmpFyAI&pageid=g38bb68a532_0_222)
+
+Now, when you look at the top of this object, you'll see that `month` and `day` remain in the data frame and that variable combines information from the other columns in airquality (`ozone`, `solar.r`, `wind`, `temp`). This is still a long format dataset; however, it has used `month` and `day` as IDs when reshaping the data frame.
+
+##### spread()
+
+To return your long data back to its original form, you can use `spread()`. Here you specify two columns: the column that contains the names of what your wide data columns should be (`key=variable`) and the column that contains the values that should go in these columns (`value=value`). The data frame resulting from `spread()` will have the original information back in the wide format (again, the columns will be in a different order). But, we'll discuss how to rearrange data in the next lesson!
+
+```r
+## use gather() to reshape from wide to long
+spread_data <- spread(gathered, key=variable, value=value)
+
+## take a look at the spread data
+head(spread_data)
+
+## compare that back to the original
+head(airquality)
+```
+
+
+![spread data](https://docs.google.com/presentation/d/14msuN3MbQE6BSIaNu2ipv1-5ypgvWlxsGwn3jmpFyAI/export/png?id=14msuN3MbQE6BSIaNu2ipv1-5ypgvWlxsGwn3jmpFyAI&pageid=g38bb68a532_0_236)
+
+#### reshape2
```

Comment:
reshape2 is retired these days, so in a future PR we will need to exchange it for what is now recommended by the tidyverse: https://www.rdocumentation.org/packages/reshape2/versions/1.4.4